### PR TITLE
basic solution

### PR DESCRIPTION
**Pull request by ihorromaniuk**

### CODE REVIEW

**Review by Reastador — APPROVED** [review]

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        this.userService = userService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    public String addMovieSession(@RequestParam Long userId,
+                                                   @RequestParam Long movieSessionId) {
+        shoppingCartService.addSession(movieSessionService.get(movieSessionId),
+                userService.getById(userId));
+        return "Movie session successfully added to users shopping cart";
```
Comment (MaliukDaria) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:37]: Lets make this method void or return updated shopping cart

File: src/main/java/mate/academy/spring/validation/impl/EmailValidator.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    private static final String EMAIL_PATTERN = "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        if (email == null) {
+            return false;
+        }
+        Pattern pattern = Pattern.compile(EMAIL_PATTERN);
+        Matcher matcher = pattern.matcher(email);
+        return matcher.matches();
```
Comment (MaliukDaria) [inline src/main/java/mate/academy/spring/validation/impl/EmailValidator.java:19]:
```suggestion
          return email != null && email.matches(EMAIL_PATTERN);
```

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/controller/AuthenticationController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private final UserRequestMapper requestMapper;
+
+    public AuthenticationController(UserResponseMapper responseMapper,
+                                    UserRequestMapper requestMapper) {
+        this.responseMapper = responseMapper;
+        this.requestMapper = requestMapper;
+    }
+
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody UserRequestDto request) {
```
[the remark this answers is not in the document]
Comment (MaliukDaria) [reply src/main/java/mate/academy/spring/controller/AuthenticationController.java:25]: We still need to add @Valid annotation in front of UserRequestDto

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Review by MaliukDaria — APPROVED** [review]
Good job!

**Pull request closed without merging**